dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/extensions/JavaEESecCDIExtension.java

```diff
@@ -333,7 +335,25 @@ private void createModulePropertiesProviderBeanForApplicationAuthMechToAdd(BeanM
                 e.printStackTrace();
             }
         }
-        addAuthMech(applicationName, implClass, implClass, props);
+
+        Class<?> existingAuthMechClass = httpAuthenticationMechanismsTracker.getExistingAuthMechClass(applicationName);
+        boolean authMechAlreadyExists = (existingAuthMechClass != null);
+        if (authMechAlreadyExists) {
+            boolean isAlternative = (alternative != null);
+            if (isAlternative) {
+                // Use the alternative instead of the existing authMech
+                httpAuthenticationMechanismsTracker.removeAuthMech(applicationName, existingAuthMechClass);
+                addAuthMech(applicationName, implClass, props);
+            } else {
+                // Do not add the AuthMech - Because one already exists and this is not an alternative.
```

Comment:
As currently coded, this extension will no longer detect when there are multiple mechanisms. Fixing as suggested in prior comment should address this.